contentcuration/contentcuration/static/js/edit_channel/import/views/SearchResults.vue

```diff
@@ -0,0 +1,156 @@
+<template>
+
+  <div class="SearchResults">
+    <div>
+      <p v-show="!resultsLoading" class="TopResults">
+        Showing top results for "{{ currentSearchTerm }}"
+      </p>
+      <button @click="goToPreviousPage()" class="button-reset BackButton">
+        Go Back To Browse
+      </button>
+    </div>
+
+    <!-- ITEM RESULTS -->
+    <div class="Results">
+      <h1 class="Results__Header">Items</h1>
+      <span v-if="resultsLoading" class="LoadingMsg">
+        Loading results for "{{ currentSearchTerm }}"...
+      </span>
+      <template v-else>
+        <div v-show="itemResults.length === 0">
+          No documents, exercises, or other files matching "{{ currentSearchTerm }}"
+        </div>
+        <ul class="list-unstyled Results__List">
+          <ImportListItem
+            v-for="item in itemResults"
+            :key="item.id"
+            :node="item"
+            :isFolder="false"
+            :isChannel="false"
+            :isRoot="true"
+            :parentIsChecked="false"
+            :store="store"
+          />
+        </ul>
+      </template>
+    </div>
+
+    <!-- TOPIC RESULTS -->
+    <div class="Results">
+      <h1 class="Results__Header">Topics</h1>
```

Comment:
It might be best to just double-import "Math Document" in your example. For the user, it would break expectations if the app de-duped their selection for them.

As for the situation where we show both "Math" and its "Math Topic" subtopic, I'm not sure if there's a clear answer. It should show up in the level-1 list of results; and for the same reasons, it might be reasonable to just import both versions of "Math Topic" to maintain the user's expectations.

Perhaps, for the sake of tidiness, we might validate the user's selections and warn them of redundant resources.